From 877010c0194ba274c4532ef304eddccd955d3378 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Wed, 28 May 2025 14:17:32 +0300
Subject: [PATCH] [JSW-799] jgs: update second image bringup status

With this patch, updated software status bits
base during image prefetch.
Firmware will start second image prefetch when
first image prefetch fails, at this point
software status base flip required.

Change-Id: I08fb9aef7c1fdf86f8d58101eb54fe791012d281
---

diff --git a/common/boot_flow.c b/common/boot_flow.c
index badee10..9d6b2cf 100644
--- a/common/boot_flow.c
+++ b/common/boot_flow.c
@@ -100,6 +100,10 @@
 
 	if (!secondary_ppboot_prefetch_started && !secondary_ppboot_prefetch_failed) {
 		curr_manifest_idx = (primary_manifest_idx + 1) % HL_FW_SPI_MANIFEST_MAX_NUM;
+
+		/* TODO: What to do if fw upd switch enabled and flow comes here? */
+		img_sts_base = PROGRESS_BRINGUP_PCI_SEC;
+
 		ret = boot_flow_prefetch_img_sub(curr_manifest_idx, img_type, &desc, img_sts_base);
 		if (!ret) {
 			secondary_ppboot_prefetch_started = true;
